app/Services/Recaptcha/GoogleRecaptchaV3Validator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Recaptcha;
+
+use GuzzleHttp\ClientInterface;
+
+final class GoogleRecaptchaV3Validator implements RecaptchaValidator
+{
+    private ClientInterface $client;
+
+    private string $secret;
+
+    public function __construct(ClientInterface $client, string $secret)
+    {
+        $this->client = $client;
+        $this->secret = $secret;
+    }
+
+    public function validate(string $token): bool
+    {
+        try {
+            $response = $this->client->request(
+                'GET',
+                'https://www.google.com/recaptcha/api/siteverify',
+                [
+                    'query' => [
+                        'secret' => $this->secret,
+                        'response' => $token,
+                    ],
+                ]
+            );
+
+            $contents = $response->getBody()->getContents();
+            $responseData = json_decode($contents, true, 5, JSON_THROW_ON_ERROR);
+
+            return ($responseData['success'] ?? false) === true;
+        } catch (\Throwable $err) {
+            return false;
```

Comment:
Ik heb het inmiddels lokaal werkend! 🎉 

Enige is dat ik nog zou willen overwegen om hier een exceptie te gooien met een bericht als "Er is een technisch probleem opgetreden bij de recaptcha validatie", met evt. de originele error message erbij. Snap dat het voor de user weinig uitmaakt (doet het niet = doet het niet), maar het maakt ons leven wel makkelijker mocht dit een keer bij iemand stuk gaan.

Reply:
Goed plan! alsjeblieft